community/kernel/src/main/java/org/neo4j/kernel/impl/api/StackingQueryRegistrationOperations.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.api;
+
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.neo4j.kernel.api.ExecutingQuery;
+import org.neo4j.kernel.impl.api.operations.QueryRegistrationOperations;
+import org.neo4j.kernel.impl.util.MonotonicCounter;
+
+public class StackingQueryRegistrationOperations implements QueryRegistrationOperations
+{
+    public static final MonotonicCounter LAST_QUERY_ID = MonotonicCounter.newAtomicMonotonicCounter();
```

Comment:
no, object creation should be fine not a problem here, just to make it as simple as it can be - i would use simple private counter for doing that, but fine with you this kind of id generation if you guys think it useful